### ISSUE TITLE
Add end_of_iteration_hook to BaseTrainer. Add hook to BaseTester. Move all record-keeper related functions to utils.logging_presets

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#3]
Replace the existing "update_records" function with a user-provided hook. In other words, replace the record_keeper argument with an input hook.

**Comment by KevinMusgrave** [comment]
In v0.9.67:

- BaseTrainer has end_of_iteration_hook, which is called at the end of each iteration, and end_of_epoch_hook, which is called at the end of each epoch.
- BaseTester has end_of_testing_hook, which is called at the end of testing.
- All mentions of record-keeper have been removed from these two classes. If you'd still like to use record-keeper, you can use the HookContainer class defined in utils.logging_presets. Here's an example:
```python
import pytorch_metric_learning.utils.logging_presets as logging_presets

pkl_folder, tensorboard_folder = "example_logs", "example_tensorboard"
record_keeper, _, _ = logging_presets.get_record_keeper(pkl_folder, tensorboard_folder)
hooks = logging_presets.get_hook_container(record_keeper)
dataset_dict = {"train": train_dataset, "val": val_dataset}

tester = testers.GlobalEmbeddingSpaceTester(end_of_testing_hook=hooks.end_of_testing_hook)
trainer = trainers.MetricLossOnly(..., 
    end_of_iteration_hook=hooks.end_of_iteration_hook,
    end_of_epoch_hook=hooks.end_of_epoch_hook(tester, dataset_dict)
)
```
See [the end of this example script](https://github.com/KevinMusgrave/pytorch-metric-learning/blob/master/examples/example_MetricLossOnly.py) for full usage.
